### ISSUE TITLE
Screen reader doesn't announce Notification

### ISSUE THREAD

**Issue by AndreasSamjeske** [vaadin/flow-components#7594]
### Description

In our webapp Chrome with NVDA and Firefox with NVDA don't announce some notifications. The behavior differs between the browser and is not reliably reproducible. Sadly, simple examples don't show the problem. Following site was of great help in understanding, what might be going on: https://tetralogical.com/blog/2024/05/01/why-are-my-live-regions-not-working/
My understanding is this: On `Notification.show("some text");` Vaadin generates this HTML in one go:
`<vaadin-notification-card aria-live="polite" role="alert" ...>some text</vaadin-notification-card>`

There are 2 problems, which interact with each other:

1. `role="alert"` shall only be used for critical event. On some platforms a screen reader announces such messages with the prefix Alert! or plays a specific earcon. That's a nice option, but shouldn't be the default in Vaadin.
2. The container with attribute aria-live has to appear in HTML first. Then later it's content has to be changed to the actual message. That's the only reliable way to trigger browser to fire the event screen readers are looking for. Granted, the ARIA spec isn't clear in this point. But apparently, this is how it is intended.

My educated guess: Vaadin adds always `role="alert"`, because this covers in part the problem of point 2. Without `role="alert"` notifications wouldn't be announced at all (with some exceptions like Firefox & JAWS).

Setting aria-live to "assertive" didn't help in our webapp. Probably because `role="alert"` is responsible for triggering the announcement right now.

### Expected outcome

- Screen readers shall always announce notifications, especially if `assertive=true`.
- `role="alert"` should be an option.

### Minimal reproducible example

I am unable to supply a simple example. But this workaround improved the reliability for notifications in our webapp:
```
        var noti = Notification.show("", DEFAULT_DURATION, DEFAULT_POSITION, true);
        UI ui = UI.getCurrent();
        ui.push();

        noti.setText(message);
```

### Steps to reproduce

Unknown.
In our webapp, Chrome has problems, when we change a lot of things in the view and at the same time create a notification. Firefox on the other hand doesn't announce a notification, when doing navigation to a different view at the same time (Notification has no parent, so it persists while changing the view).
Warning for tests: Having developer tools open in Firefox changed its behavior. I guess, that's s another hint, that this might be a timing related issue.

### Environment

Vaadin version(s): 24.7.6
OS: Windows 11 Version 23H2
NVDA 2024.4.2
Chrome Version 137.0.7151.104
Firefox 139.0.4 (64-Bit)

### Browsers

Chrome, Firefox

**Comment by web-padawan** [comment]
Thanks for the issue. There is a `setAssertive()` method and a constructor that allows you to configure Notification to be assertive:

https://github.com/vaadin/flow-components/blob/f09a804b55fe959b1b2d17ef27d89da6ee86674e/vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java#L220-L221

https://github.com/vaadin/flow-components/blob/f09a804b55fe959b1b2d17ef27d89da6ee86674e/vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java#L275-L276

Maybe we could also add a separate option to not use `role="alert"` by default if setting assertive doesn't help. That needs some research.

**Comment by AndreasSamjeske** [comment]
Thanx for your quick response.
About a year ago our testers reported to us, that some notifications don't get announced. We verified that finding, but were unable to pinpoint the problem. We suspected `aria-live="polite"` would be to blame. But on the other hand, Vaadin also added `role="alert"` to the notification container. That should do the job, was our understanding back then.
Recently, we discovered, that Vaadin Notification offers the (new?) method `setAssertive`. We gave it a shot: All our notifications were set to be assertive and hence having the attribute `aria-live="assertive"`. The outcome was like before: All notifications, that were previously not announced, still failed to be announced. 😞
That lead to the deep dive I took into ARIA spec of aria-live and how browser shall handle it. In essence:

1. `setAssertive(true)` doesn't improve the situation. At least not yet.
2. The workaround (Creating empty notification, UI push, setText or add(..) to set content of notification) fixes the problem.

My not tested guesses regarding `role="alert"`:
1. Removing `role="alert"` would worsen the situation. But the workaround would still fix it. Or the other way around: With the workaround, `role="alert"` is no longer necessary.
2. Without `role="alert"` I suspect `setAssertive(true)` having an impact: Polite notifications might not be announced, if the screen reader has more important things to say. Assertive notifications would disrupt the speech flow, since they have higher priority.

Conclusion:
Vaadin has to implement a behavior like I did in the workaround, since the ARIA spec on aria-live states, that this is the way to go.

**Comment by rolfsmeds** [comment]
I think a more reliable fix than emulating the _add-to-DOM + push + set-text_ workaround described above would be to have an always-present notification container in the DOM, which would be the live region, and always generate notification instances inside it.

Or perhaps two separate containers would be needed to support both `assertive` and `polite.`

A bonus of having notifications in such a container is that it could help solve their keyboard-reachability by providing a single target for a dedicated keyboard shortcut that [moves focus to notifications](https://github.com/vaadin/web-components/issues/602), and within which all currently open notifications could be focus-traversed as if they were within the normal UI flow.
